### DISCUSSION TITLE
Wrong fullyQualifiedName

### DISCUSSION THREAD

**Discussion by aurelcarlier** [LemonAppDev/konsist#829] [Report a Problem]
### Tell us what version of Konsist you use.

0.13.0

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hi,

I've found out a wrong return value for the fullyQualifiedName property when the return type of a given function ends with the same suffix of another class also imported in the file. Here is a minimal code that hightight this type confusion.

### Actual behaviour

Let's consider two packages `poc` and `other` with the following code:

```kotlin
package poc

import com.lemonappdev.konsist.api.Konsist
import com.lemonappdev.konsist.api.ext.list.print
import com.lemonappdev.konsist.api.ext.list.withReturnType
import other.AnotherShape

class Bob {

    fun foo(): Shape {
        return Shape()
    }

    fun bar(): AnotherShape {
        return AnotherShape()
    }
}

class Shape

fun main() {
    Konsist.scopeFromPackage("poc").functions().withReturnType().print {
        "${it.name}() returns type '${it.returnType!!.name}' with fully qualified name '${it.returnType!!.fullyQualifiedName}'"
    }
}
```

```kotlin
package other

class AnotherShape
```

### Expected behaviour

Running the `main` function gives the following output:

```bash
foo() returns type 'Shape' with fully qualified name 'other.AnotherShape' => WRONG, it should be 'poc.Shape'
bar() returns type 'AnotherShape' with fully qualified name 'other.AnotherShape' => OK
```

**Comment by jibidus** [comment]
It seems a duplicate of #738

**Reply by jibidus** [reply]
ℹ️  This issue is not resolved in `develop` branch at this time (tested locally).

**Reply by igorwojda** [reply]
Fixed in https://github.com/LemonAppDev/konsist/pull/830

**Comment by nataliapeterwas** [comment]
A new version v0.14.0 has been released. It contains the fix.
See [v0.14.0 release](https://github.com/LemonAppDev/konsist/releases/tag/v0.14.0)
